src/common/allocator.h

```diff
@@ -4,7 +4,11 @@
 #ifndef CEPH_COMMON_ALLOCATOR_H
 #define CEPH_COMMON_ALLOCATOR_H
 
+#if defined(__linux__)
 #include <malloc.h>
+#else
+#include <stdlib.h>
```

Comment:
@dillaman can we s/memalign/aligned_alloc/ ? we can implement `allocate()` like
```c++
return static_cast<T*>(
  aligned_alloc(alignof(T), round_up_to(alignof(T), n * sizeof(T))));
```

as the former is deprecated. and i don't think that tcmalloc is using it. instead it implements it using mmap, hugepage or `/dev/mem`.

what do you think?

Reply:
The bug is that tcmalloc is missing `aligned_alloc` on several distro, so you cannot use it if ```LIBTCMALLOC_MISSING_ALIGNED_ALLOC```

Reply:
The fixed versions of tcmalloc just alias `aligned_alloc` to its internal `tc_memalign` call [1].

[1] https://github.com/alk/gperftools/commit/d7e9cbf4ea96fed72af5cbb996c9690a23ecaa4b

Reply:
d'oh! sorry, i forgot/missed the whole point why you implemented this the first place! thanks for the explanation!